src/Appwrite/Utopia/Response/Model/Variable.php

```diff
@@ -41,11 +41,17 @@ public function __construct()
                 'default' => '',
                 'example' => 'myPa$$word1',
             ])
-            ->addRule('functionId', [
+            ->addRule('resourceType', [
                 'type' => self::TYPE_STRING,
-                'description' => 'Function ID.',
+                'description' => 'Service to which the variable belongs. Possible values are "project", "function"',
                 'default' => '',
-                'example' => '5e5ea5c16897e',
+                'example' => 'function',
+            ])
+            ->addRule('resourceId', [
+                'type' => self::TYPE_STRING,
+                'description' => 'ID of resource to which the variable belongs. If resourceType is "project", it is empty. If resourceType is "function", it is ID of the function.',
```

Comment:
if the resource type is project, shouldnt the resource ID be the project ID ? 👀

Reply:
Empty, in that case. Because this is collection in dbForProject. We are already project-scoped.